service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerDataRepoSnapshotConnectedTest.java

```diff
@@ -134,6 +141,49 @@ public void create() throws Exception {
     assertApiDataRepoEquals(sourceResource, gotResource);
   }
 
+  @Test
+  public void create_withSnapshotHavingGroupConstraint() throws Exception {
+    logger.info("features.isTpsEnabled(): %s".formatted(features.isTpsEnabled()));
+    // Don't run the test if TPS is disabled
+    if (!features.isTpsEnabled()) {
+      return;
+    }
+
+    // Clean up policies from previous runs, if any exist
+    mockWorkspaceV1Api.deletePolicies(userAccessUtils.defaultUserAuthRequest(), workspaceId);
```

Comment:
Ostensibly because of the flag, but in practice when the flag is off the tests universally fail for this class due to the nature of the cleanup method. I'll move these to the cleanup method.